File: solutions/csharp/high-scores/1/HighScores.cs
using System.Collections.Generic;
using System.Linq;

public class HighScores
{
	private readonly List<int> _list;

	public HighScores(List<int> list) =>
		_list = list;

	public List<int> Scores() =>
		_list;

	public int Latest() =>
		_list.Last();

	public int PersonalBest() =>
		_list.Max();

	public List<int> PersonalTopThree()
	{
		List<int> listCopy = new(_list);
		listCopy.Sort((a, b) => b.CompareTo(a));
		return listCopy.Take(3)
			.ToList();
	}
}

File: solutions/java/high-scores/1/src/main/java/HighScores.java
import java.util.Comparator;
import java.util.List;
import java.util.stream.Collectors;

record HighScores(List<Integer> highScores) {

	List<Integer> scores() {
		return this.highScores;
	}

	Integer latest() {
		return this.highScores.getLast();
	}

	Integer personalBest() {
		return this.highScores.stream()
				       .max(Integer::compareTo)
				       .orElseThrow();
	}

	List<Integer> personalTopThree() {
		return this.highScores.stream()
				        .sorted(Comparator.reverseOrder())
				        .limit(3)
				        .collect(Collectors.toList());
	}
}
